pkg/operator/ceph/cluster/mgr/spec.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright 2018 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mgr
+
+import (
+	"fmt"
+	"strconv"
+
+	mgrdaemon "github.com/rook/rook/pkg/daemon/ceph/mgr"
+	opmon "github.com/rook/rook/pkg/operator/ceph/cluster/mon"
+	opspec "github.com/rook/rook/pkg/operator/ceph/spec"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	"k8s.io/api/core/v1"
+	extensions "k8s.io/api/extensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	mgrDaemonCommand = "/usr/bin/ceph-mgr"
```

Comment:
I feel like full path is a pattern I have seen on a lot of containers. I think it is preferred because some entrypoints aren't able to resolve the short command; this is a situation I have encountered myself. The Rook container's entrypoint is tini, which in my testing does call the binary if just `ceph-mgr` is used. But the Ceph containers don't have an entrypoint specified, which IIRC won't interpret `ceph-mgr` to be `/usr/bin/ceph-mgr`.

We could solve the same issue by specifying the entrypoint as `['/usr/bin/sh', '-c']` potentially. But then, building a custom container with a custom entrypoint to do extra debugging/logging could be something someone would want to do? I'm really not sure.

Reply:
in my testing with the ceph/ceph container and pulling in your changes, I actually found that I needed to just specify `ceph-mgr` instead of including the full path. By running the daemon directly, i believe we don't have to worry about pid 1 issues either since no subprocesses are started

Reply:
I looked around and saw that for `entrypoint`, usage seems to always be full path. But for command, I saw examples of just the command itself. I can see some pro/cons for choosing one over the other, but I think biggest pro for being able to use `ceph-mgr` is that the user can build custom containers where the binary is not in `/usr/bin` if they are feeling adventurous/lucky.